lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -6803,6 +6803,37 @@ class DecomposeAtenReshapeAsOp : public OpRewritePattern<AtenReshapeAsOp> {
 };
 } // namespace
 
+namespace {
+// Decompose AtenLinalgNormOp to AtenLinalgVectorNormOp only
+class DecomposeAtenLinalgNormOp : public OpRewritePattern<AtenLinalgNormOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenLinalgNormOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+
+    if (op.getDim().getDefiningOp()) {
+      auto dimOp =
+          dyn_cast<Torch::PrimListConstructOp>(op.getDim().getDefiningOp());
+      if (!dimOp || llvm::to_vector(dimOp.getElements()).size() != 1)
+        return rewriter.notifyMatchFailure(
+            op, "Unimplemented: only dim size of 1 is supported");
+    } else
+      return rewriter.notifyMatchFailure(op, "expect dim param to be a list");
+
+    // default ord value is 2 for vector_norm
+    auto ord = op.getOrd();
+    if (dyn_cast_or_null<ConstantNoneOp>(ord.getDefiningOp())) {
```

Comment:
nit: to avoid relying on `getDefiningOp`, you can do `ord.getType().isa<Torch::NoneType>()`